### ISSUE TITLE
Media source paths should not be inserted into the TV values table.

### ISSUE THREAD

**Issue by BigBlockStudios** [modxcms/revolution#12852]
The 'Image TV' type inserts the image into the TV value database table WITH the entire mediasource path/base.

The media source path/base should be a variable and prepended to media when the page is rendered.

The specific issue that arises is:

- you can't relocate the media source to a different directory [without breaking all your image based TVs]
- you can't change the media source baseUrl after image TV values have been populated [without breaking all your image based TVs] 

i.e. 
- you create a media source /usermedia/ 
- populate a whole bunch of TVs
- the value would be saved as /usermedia/_path_/image.jpg
- change the baseUrl /usermedia/ to be compatible with something like oooo say redactor  
- all your images will be output in the your resources as /usermedia/usermedia/_path_/image.jpg

Basically modx is prepending the mediasource baseUrl when the page renders anyway - it should be treated as a variable & not inserted into the TV values table.

-thanks
-sean

**Comment by sottwell** [comment]
The only time I know of where it does that is when the Media Source URL is not relative.

**Comment by sottwell** [comment]
TVs have a "source" field to specify the media source, and the contentvalue will only have the part of the file path that is not in the media source. So if you have a media source that has "assets/images/" in its path and URL, but within that source you have further folders, the final path to the image will have everything after assets/images/.

**Comment by BigBlockStudios** [comment]
Sure, ok, but basically if it is ~sometimes~ treated as a variable it should ALWAYS be treated as a variable.  
Compound that with not being able to move a media source directory once it has been created

**Comment by sottwell** [comment]
I don't understand your problem. If your Media Source is set up correctly, of course you can move it. The Media Source path and URL aren't saved in the TV. Only the parts that aren't in the Media Source are saved in the TV.

**Comment by sottwell** [comment]
I think this is more an issue with implementation, not an issue. You should move this to the forums or a Slack channel for help.

**Comment by BigBlockStudios** [comment]
Looking into the tv values table 
![usermedia](https://cloud.githubusercontent.com/assets/779531/12331248/c1f328ba-bab6-11e5-876c-694f0382724a.gif)

all the image tvs were added with the /usermedia....

**Comment by adamwintle** [comment]
You could use a System Setting as a variable for this in your templates, so you could have `[[++assets_path]]/[[++user_media_path]]/banners/files.jpg` there's lots of ways around this using MODX. 

I agree with @sottwell this isn't a MODX core bug and more of a setup problem, we're all on the Slack chat at http://modx.org all the time, feel free to join a chat room.

@modxbot close
